clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLWriter.java

```diff
@@ -141,6 +142,64 @@ public void writeContent(XMLExtendedStreamWriter writer, SubsystemMarshallingCon
                     writer.writeEndElement();
                 }
             }
+
+            if (model.hasDefined(RemoteCacheContainerResourceDefinition.WILDCARD_PATH.getKey())) {
+                for (Property entry : model.get(RemoteCacheContainerResourceDefinition.WILDCARD_PATH.getKey()).asPropertyList()) {
+
+                    String remoteContainerName = entry.getName();
+                    ModelNode remoteContainer = entry.getValue();
+
+                    writer.writeStartElement(XMLElement.REMOTE_CACHE_CONTAINER.getLocalName());
+                    writer.writeAttribute(XMLAttribute.NAME.getLocalName(), remoteContainerName);
+
+                    writeAttributes(writer, remoteContainer, EnumSet.allOf(RemoteCacheContainerResourceDefinition.Attribute.class));
+
+                    if (remoteContainer.hasDefined(ClientThreadPoolResourceDefinition.WILDCARD_PATH.getKey())) {
+                        writeThreadPoolElements(XMLElement.ASYNC_THREAD_POOL, ClientThreadPoolResourceDefinition.ASYNC, writer, remoteContainer);
+                    }
+
+                    if (remoteContainer.hasDefined(ConnectionPoolResourceDefinition.PATH.getKeyValuePair())) {
+                        ModelNode connectionPool = remoteContainer.get(ConnectionPoolResourceDefinition.PATH.getKeyValuePair());
+                        final EnumSet<ConnectionPoolResourceDefinition.Attribute> attributes = EnumSet.allOf(ConnectionPoolResourceDefinition.Attribute.class);
+                        if (hasDefined(connectionPool, attributes)) {
+                            writer.writeStartElement(XMLElement.CONNECTION_POOL.getLocalName());
+                            writeAttributes(writer, connectionPool, attributes);
+                            writer.writeEndElement();
+                        }
+                    }
+
+                    if (remoteContainer.hasDefined(InvalidationNearCacheResourceDefinition.PATH.getKeyValuePair())) {
+                        writer.writeStartElement(XMLElement.INVALIDATION_NEAR_CACHE.getLocalName());
+                        ModelNode nearCache = remoteContainer.get(InvalidationNearCacheResourceDefinition.PATH.getKeyValuePair());
+                        writeAttributes(writer, nearCache, EnumSet.allOf(InvalidationNearCacheResourceDefinition.Attribute.class));
+                        writer.writeEndElement();
+                    }
+
+//                    if (remoteContainer.hasDefined(RemoteClustersResourceDefinition.PATH.getKeyValuePair())) {
+//                        ModelNode remoteClusters = remoteContainer.get(RemoteClustersResourceDefinition.PATH.getKeyValuePair());
+//                        writeAttributes(writer, remoteClusters, EnumSet.allOf(RemoteClustersResourceDefinition.Attribute.class));
+
+                        if (remoteContainer.hasDefined(RemoteClusterResourceDefinition.WILDCARD_PATH.getKey())) {
```

Comment:
There will always be at least 1 remote cluster, so this check is unnecessary.

Reply:
Right, the original idea was to support the collocated cluster. Removed now.